api/v1alpha1/nodepool_types.go

```diff
@@ -211,3 +214,33 @@ type Filter struct {
 	// Values includes one or more filter values. Filter values are case-sensitive.
 	Values []string `json:"values"`
 }
+
+// Volume encapsulates the configuration options for the storage device
+type Volume struct {
```

Comment:
Update based on review feedback: @taragu agreed to remove `DeviceName,Encrypted,EncryptionKey` as these are not required for the desired use case.

We will leave `Type` as a string for now.